File: app/models/course.go
package models

import (
	"github.com/goravel/framework/database/orm"
)

type Course struct {
	orm.Model

	CategoryID  uint   `json:"category_id" form:"category_id"`
	Title       string `json:"title" form:"title"`
	Description string `json:"description" form:"description"`
	CoverPath   string `json:"cover_path" form:"cover_path"`
	Status      uint8  `json:"status" form:"status"`
}

File: src/types/course.ts
/**
 * 课程数据
 */
export type CourseResource = {
  /**
   * 课程ID
   */
  id: number;
  /**
   * 课程标题
   */
  title: string;
  /**
   * 简介
   */
  description: string;
  /**
   * 封面路径
   */
  cover_path: string;
  /**
   * 创建日期
   */
  created_at: string;
};
